### ISSUE TITLE
Feature request: Add `--generate-ignore` flag

### ISSUE THREAD

**Issue by philderbeast** [ndmitchell/hlint#1374]
When a large project has not been linted before it can be quite a bit of work to tackle and I'd prefer to do this a suggestion at a time. Towards that end, it would be helpful if we could generate an `.hlint.yaml` file for a clean pass with comments showing the number of occurrences. Something like this (from [liquid-fixpoint](https://github.com/ucsd-progsys/liquid-fixpoint)):

```yaml
# .hlint.yaml
# Lint hints found with hlint-3.3.6
# All supressed with the following ignores.
- ignore: {name: Redundant bracket} # 198 hints
- ignore: {name: Redundant if} # 1 hint
- ignore: {name: Redundant guard} # 2 hints
- ignore: {name: Redundant lambda} # 3 hints
- ignore: {name: Redundant $} # 20 hints
- ignore: {name: Redundant <$>} # 29 hints
- ignore: {name: Redundant bang pattern} # 33 hints
- ignore: {name: Redundant flip} # 1 hint
- ignore: {name: Redundant return} # 1 hint
- ignore: {name: Use camelCase} # 2 hints
- ignore: {name: Use newtype instead of data} # 3 hints
- ignore: {name: Use section} # 3 hints 
- ignore: {name: Use gets} # 9 hints
- ignore: {name: Use guards} # 3 hints
- ignore: {name: Use maybe} # 7 hints
- ignore: {name: Use fold} # 1 hint
- ignore: {name: Use map} # 2 hints
- ignore: {name: Use map once} # 1 hint
- ignore: {name: Use mapMaybe} # 3 hints
- ignore: {name: Use concatMap} # 2 hints
- ignore: {name: Use infix} # 3 hints
- ignore: {name: Use elem} # 1 hint
- ignore: {name: Use notElem} # 2 hints
- ignore: {name: Use <$>} # 10 hints
- ignore: {name: Use <$} # 2 hints
- ignore: {name: Use $>} # 7 hints
- ignore: {name: Use if} # 1 hint
- ignore: {name: Use /=} # 1 hint
- ignore: {name: Use ++} # 1 hint
- ignore: {name: Use const} # 1 hint
- ignore: {name: Use void} # 2 hints
- ignore: {name: Use first} # 1 hint
- ignore: {name: Use second} # 1 hint
- ignore: {name: Use all} # 2 hints
- ignore: {name: Use tuple-section} # 1 hint
- ignore: {name: Use record patterns} # 5 hints
- ignore: {name: Eta reduce} # 30 hints
- ignore: {name: Fix pragma markup} # 41 hints
- ignore: {name: Functor law} # 3 hints
- ignore: {name: Avoid lambda} # 4 hints
- ignore: {name: Move brackets to avoid $} # 8 hints
- ignore: {name: Evaluate} # 1 hint
```

Running like normal, the `--generate-ignore` flag would update `.hlint.yaml` so that the next hlint run would result in 0 hints.

**Comment by philderbeast** [comment]
This could also operate like the `--default` flag and print to stdout.

```
# To generate a suitable file for HLint do:
$ hlint --default > .hlint.yaml
```

**Comment by philderbeast** [comment]
The `--default` **"Print a default file to stdout"** option does this (except for commenting on the number of hits). There's a good description of a hlint workflow with `--default` in the readme. 

```markdown
### Suggested usage

HLint usage tends to proceed in three distinct phases:

1. Initially, run `hlint . --report` to generate `report.html` containing a list
of all issues HLint has found. Fix those you think are worth fixing and keep
repeating.
2. Once you are happy, run `hlint . --default > .hlint.yaml`, which will
generate a settings file ignoring all the hints currently outstanding. Over time
you may wish to edit the list.
3. For larger projects, add [custom hints or rules](./README.md#customizing-the-hints).
```

This feature of hlint is easy to miss in its description on the command line. I was looking for an option more like `--willful-ignorance` **"Suppress all found hints"**.
